### DISCUSSION TITLE
Why it is aiming at localhost, i already setup MEDUSA_BACKEND_URL in prod. Admin panel

### DISCUSSION THREAD

**Discussion by ArnavK-09** [medusajs/medusa#6151] [Q&A]
![Screenshot_2024-01-21-22-23-51-22_4d38fce200f96aeac5e860e739312e76.jpg](https://github.com/medusajs/medusa/assets/69188140/ac7bccb7-507b-415d-9806-74f06eb78a4b)

**Comment by kevinzhang2012** [comment]
```
  {
    resolve: `@medusajs/file-local`,
    options: {
      upload_dir: "uploads",
      backend_url: process.env.MEDUSA_BACKEND_URL,
    },
  },
```
If it is previously uploaded asset you will need to re-upload, since the asset URL has been recorded to DB.

**Reply by ArnavK-09** [reply]
Still :/

![Screenshot_2024-01-28-00-55-14-44_4d38fce200f96aeac5e860e739312e76.jpg](https://github.com/medusajs/medusa/assets/69188140/9eda2aee-02c5-4222-ab4e-5659c8d1fda5)

**Comment by ArnavK-09** [comment]
@hosenur I used the **cloudinary** plugin instead it is working now

**Reply by ArnavK-09** [reply]
![Screenshot_2024-02-01-00-42-43-84_e307a3f9df9f380ebaf106e1dc980bb6.jpg](https://github.com/medusajs/medusa/assets/69188140/e99e6e68-2b89-4354-962a-1c85b38958df)

**Reply by hosenur** [reply]
Thank you , It seems the URLs of previously added products were generated with the localhost URL, on updating the `backend_url` for the plugin and then deleting all assets and re uploading them fixes it as now it generated the asset URLs with the updated endpoint.

**Comment by danielgran** [comment]
IMHO this is a bug. the url should not be stored fully.
The reason i see is that medusa is intended to be ran on many different hosts, which possibly are used to preconfigure a production environment. This concept works BUT the images.

So either migrate on `media` Table or reupload.
